### PR TITLE
Pre-release handling

### PR DESCRIPTION
**Pull request by jburel**
Adjust support for cases like 5.6.x-y 
It will return 5.6.x-y, 5.6.x, 5.6, 5 and latest (assuming it is the latest in that example).
cc @manics

### CODE REVIEW

File: __tests__/main.test.js [hunk 1]
```diff
@@ -184,3 +225,56 @@ test("Branch", async () => {
...
+      },
+      {
+        name: "1.1.0-0",
+      },
+    ]);
+  const tags = await calculateTags(
+    "TOKEN",
+    "owner",
+    "repo",
+    "refs/tags/1.1.0-1",
```
Comment (consideRatio) [inline __tests__/main.test.js:275]: Hi @jburel! :D I very much appreciate this github action and that it has been allowed to migrate to the JupyterHub org! Thank you for your contributions to it!!!

---

The git reference that triggers the build is tested to be a build version later than those available in the tests above. What do you think about adding one for when you have a the tag `1.0.0` and `1.0.0-2` and then want to calculate the tags from `1.0.0-1`?

I expect our logic could be sensitive to malfunctioning in those situations.

**Review by manics — APPROVED** [review]
Thanks!

File: __tests__/main.test.js (hunk 1 above)
Comment (jburel) [reply __tests__/main.test.js:275]: I assume that in that case, you expect to only have ``1.0.0-1`` as the returned value

Comment (consideRatio) [reply __tests__/main.test.js:275]: Yes, do you think that makes sense?

I just spent some time considering this further and made a PR to this PR: https://github.com/jburel/action-major-minor-tag-calculator/pull/1

**Comment by consideRatio** [comment]
Wiee thank you for your work on this @jburel!

**Comment by consideRatio** [comment]
Ah I now see @manics have already given you write access to this repo @jburel - great!

**Pull request merged**